File: AutoRest/Generators/Python/Python.Tests/AcceptanceTests/bool_tests.py
import unittest
import subprocess
import sys
import isodate
import tempfile
import json
from datetime import date, datetime, timedelta
import os
from os.path import dirname, pardir, join, realpath, sep, pardir

cwd = dirname(realpath(__file__))
root = realpath(join(cwd , pardir, pardir, pardir, pardir, pardir))
sys.path.append(join(root, "ClientRuntimes" , "Python", "msrest"))

tests = realpath(join(cwd, pardir, "Expected", "AcceptanceTests"))
sys.path.append(join(tests, "BodyBoolean"))

from msrest.serialization import Deserializer
from msrest.exceptions import DeserializationError

from auto_rest_bool_test_service import (
    AutoRestBoolTestService, 
    AutoRestBoolTestServiceConfiguration)

from auto_rest_bool_test_service.models import ErrorException

class BoolTests(unittest.TestCase):

    def test_bool(self):

        config = AutoRestBoolTestServiceConfiguration("http://localhost:3000")
        config.log_level = 10
        client = AutoRestBoolTestService(config)

        self.assertTrue(client.bool_model.get_true())
        self.assertFalse(client.bool_model.get_false())

        client.bool_model.get_null()
        client.bool_model.put_false(False)
        client.bool_model.put_true(True)

        with self.assertRaises(ErrorException):
            client.bool_model.put_true(False)

        with self.assertRaises(DeserializationError):
            client.bool_model.get_invalid()

if __name__ == '__main__':
    unittest.main()

File: src/generator/AutoRest.Java.Tests/src/test/java/fixtures/bodyboolean/BoolTests.java
package fixtures.bodyboolean;

import com.fasterxml.jackson.core.JsonParseException;

import org.junit.Assert;
import org.junit.BeforeClass;
import org.junit.Test;

import fixtures.bodyboolean.implementation.AutoRestBoolTestServiceImpl;

import static org.junit.Assert.fail;

public class BoolTests {
    private static AutoRestBoolTestService client;

    @BeforeClass
    public static void setup() {
        client = new AutoRestBoolTestServiceImpl("http://localhost:3000");
    }

    @Test
    public void getNull() throws Exception {
        try {
            client.bools().getNull();
            fail();
        } catch (NullPointerException e) {
            // expected
        }
    }

    @Test
    public void getInvalid() throws Exception {
        try {
            client.bools().getInvalid();
            Assert.assertTrue(false);
        } catch (Exception exception) {
            // expected
            Assert.assertEquals(JsonParseException.class, exception.getCause().getClass());
        }
    }

    @Test
    public void getTrue() throws Exception {
        boolean result = client.bools().getTrue();
        Assert.assertTrue(result);
    }

    @Test
    public void getFalse() throws Exception {
        boolean result = client.bools().getFalse();
        Assert.assertFalse(result);
    }

    @Test
    public void putTrue() throws Exception {
        client.bools().putTrue(true);
    }

    @Test
    public void putFalse() throws Exception {
        client.bools().putFalse(false);
    }
}
